src/domain/data-store/DataStoreMetadata.ts

```diff
@@ -0,0 +1,101 @@
+import _ from "lodash";
+import { Maybe } from "../../types/utils";
+import { MetadataImportParams } from "../metadata/entities/MetadataSynchronizationParams";
+import { ObjectSharing } from "../storage/repositories/StorageClient";
+
+export type DataStoreNamespace = string;
+export type DataStoreKey = { id: string; value: any };
+
+export type DataStoreAttrs = { namespace: DataStoreNamespace; keys: DataStoreKey[]; sharing: Maybe<ObjectSharing> };
+
+export class DataStoreMetadata {
+    public readonly namespace: DataStoreNamespace;
+    public readonly keys: DataStoreKey[];
+    public readonly sharing: Maybe<ObjectSharing>;
+
+    static NS_SEPARATOR = "[NS]";
```

Comment:
This was my first option, but you can put "/" in a namespace.

![image](https://github.com/user-attachments/assets/33918bc9-8b9a-4fa8-9aa4-b73bbc68fbd7)

Reply:
Wow, it's weird that DHIS2 allows to create a namespace, but I think it's just a glitch ("ns/something" is not allowed). When you reload the data store app, "ns/" appears as "ns". 

It could not be any other way, as endpoints have the form `/api/dataStore/NAMESPACE/KEY`

Reply:
Wow, it's weird that DHIS2 allows to create a namespace, but I think it's just a glitch ("ns/something" is not allowed). When you reload the data store app, "ns/" appears as "ns". 

It could not be any other way, as endpoints have the form `/api/dataStore/NAMESPACE/KEY`